### PR TITLE
Xr data array asdf

### PR DESCRIPTION
**Pull request by vhirtham**
Contains draft versions to serialize `xarray.DataArray`, `xarray.Dataset` and `weldx.transformations.LocalCoordinateSystem`

### CODE REVIEW

File: tests/asdf/test_core.py [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+    dax = xr.DataArray(data=data, dims=["d1", "d2"], coords=coords)
+    return dax
+
+
+def test_xarray_data_array_save():
+    """Test if an xarray.DataArray can be written to an asdf file."""
+    dax = get_xarray_example_data_array()
+    tree = {"dax": dax}
+    with asdf.AsdfFile(tree, extensions=[WeldxExtension(), WeldxAsdfExtension()]) as f:
+        f.write_to("xarray.asdf")
```
Comment (CagtayFabry) [inline tests/asdf/test_core.py:32]: write to/read from io.BytesIO buffers in tests

File: weldx/asdf/tags/weldx/core/netcdf.py [hunk 2]
```diff
@@ -0,0 +1,104 @@
+import numpy as np
+from pandas.api.types import is_datetime64_any_dtype as is_datetime
+
+from weldx.asdf.types import WeldxType
+
+
+class NetCDFAttribute:
+    def __init__(self, name, data):
+        self.name = name
+        self.data = data
```
Comment (CagtayFabry) [inline weldx/asdf/tags/weldx/core/netcdf.py:10]: This looks like a perfect `@dataclass` application 😉 
(same for other NetCDF classes)

File: tests/asdf/test_core.py [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+        },
+        coords={
+            "lon": (["x", "y"], lon),
+            "lat": (["x", "y"], lat),
+            "time": pd.date_range("2014-09-06", periods=3),
+            "reference_time": pd.Timestamp("2014-09-05"),
+        },
+    )
+
+    return dsx
```
Comment (CagtayFabry) [inline tests/asdf/test_core.py:70]: maybe expand the example with `attrs` dictionary on the Dataset as well as individual DataArrays (simple comments)

Also string coordinates would be interesting here (any other special dtypes we need to look out for?)

File: weldx/asdf/schemas/weldx.bam.de/weldx/core/netcdf/attribute-1.0.0.yaml [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    type: string
+  data:
+    type: array
+
+
+required: [name, data]
+propertyOrder: [name, data]
+
+flowStyle: block
+additionalProperties: true
```
Comment (CagtayFabry) [inline weldx/asdf/schemas/weldx.bam.de/weldx/core/netcdf/attribute-1.0.0.yaml:23]: default to `false`

File: weldx/asdf/schemas/weldx.bam.de/weldx/core/netcdf/variable-1.0.0.yaml [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+description: |
+  TODO: DESCRIPTION
+type: object
+properties:
+  name:
+    type: string
+  dimensions:
+    type: array
+  dtype:
+    type: string
```
Comment (CagtayFabry) [inline weldx/asdf/schemas/weldx.bam.de/weldx/core/netcdf/variable-1.0.0.yaml:18]: add description for dtype style referencing numpy documentation

File: weldx/asdf/schemas/weldx.bam.de/weldx/core/xarray/data_array-1.0.0.yaml [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+  variables:
+      $ref: "tag:weldx.bam.de:weldx/core/netcdf/variable-1.0.0"
+
+required: [attributes, coordinates, variables]
+propertyOrder: [attributes, coordinates, variables]
+
+
+
+flowStyle: block
+additionalProperties: true
```
Comment (CagtayFabry) [inline weldx/asdf/schemas/weldx.bam.de/weldx/core/xarray/data_array-1.0.0.yaml:34]: `false`

File: tests/asdf/test_core.py (hunk 3 above)
Comment (vhirtham) [reply tests/asdf/test_core.py]: Added a simple dictionary as attributes in the tests. We still have to consider the string coordinates.

**Pull request merged**